### DISCUSSION TITLE
Planning a Trajectory

### DISCUSSION THREAD

**Discussion by GideonLombardo** [USC-ACTLab/crazyswarm#688] [Q&A]
Hello,

I am looking to define and upload a trajectory for my Crazyflies. I have read through the documentation, but I was wondering if there were more concrete examples on defining a trajectory, uploading it, and running it. Please let me know if these exists.

Thank you,
Gideon

**Comment by whoenig** [comment]
Have you seen the slides 30 to 33 at https://www.bitcraze.io/about/events/documents/bam2021/hoenig_crazyswarm_bam2021.pdf? These are not part of the readthedocs, yet, but perhaps already sufficient for you.

**Reply by GideonLombardo** [reply]
Thank you, the slides are very helpful. I am still confused about how the csv files get generated for the trajectories. Does this happen when the trajectory gets defined? I see this for circle_with_time.py: 

```
import numpy as np

numRobots = 5

r = 0.3
height = 0.7
w = 2 * np.pi / numRobots
T = 2 * 2 * np.pi / w

# horizontal circles
for i in range(0, numRobots):
    phase = 2 * np.pi / numRobots * i

    with open("timed_waypoints_circle{}.csv".format(i), "w") as f:
        f.write("t,x,y,z,yaw\n")

        for t in np.linspace(0, T, 100):
            f.write("{},{},{},{},{}\n".format(t, r * np.cos(w * t + phase), r * np.sin(w * t + phase), height, 0)) 
```

So I believe it is necessary to write to a csv file in the trajectory script. Am I correct in assuming this?

**Reply by whoenig** [reply]
It's an offline method: You first write a csv file manually (e.g., waypoints, or timed waypoints). Then, you generate the csv-file that is needed for the Crazyflie (which requires 8th-order polynomials) using the https://github.com/whoenig/uav_trajectories repo. Online, you upload this last csv file before takeoff and start it after takeoff.

**Reply by GideonLombardo** [reply]
Thank you for this information!
